### PR TITLE
cnf-tests: Add metallb feature configs

### PR DESCRIPTION
**Pull request by mmirecki**

### CODE REVIEW

File: feature-configs/deploy/metallb/operator_namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: metallb-system
```
Comment (SchSeba) [inline feature-configs/deploy/metallb/operator_namespace.yaml:4]: nit: new line please

File: feature-configs/deploy/metallb/operator_operatorgroup.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: metallb-operator
+  namespace: metallb-system
+spec:
+  targetNamespaces:
+  - metallb-system
```
Comment (SchSeba) [inline feature-configs/deploy/metallb/operator_operatorgroup.yaml:8]: nit: new line please

File: feature-configs/deploy/metallb/operator_subscription.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: metallb-operator-sub
+  namespace: metallb-system
+spec:
+  name: metallb-operator
+  channel: alpha
+  source: metallb-operator
+  sourceNamespace: metallb-system
```
Comment (SchSeba) [inline feature-configs/deploy/metallb/operator_subscription.yaml:10]: nit: new line please

File: feature-configs/deploy/metallb/operator_catalogsource.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: metallb-operator
+  namespace: metallb-system
+spec:
+  displayName: MetalLB Operator Upstream
+  # Change this to registry.redhat.io/... once metallb operator is released
+  image: quay.io/mmirecki/metallb-operator-index
```
Comment (SchSeba) [inline feature-configs/deploy/metallb/operator_catalogsource.yaml:9]: to what operator version this index point?

Is the openshift/metallb-operator or the u/s one?

**Review by SchSeba — COMMENTED** [review]
General question is the tests are able to run in the cloud?

Did you test it on a baremetal env?

File: feature-configs/deploy/metallb/operator_catalogsource.yaml (hunk 4 above)
Comment (sabinaaledort) [reply feature-configs/deploy/metallb/operator_catalogsource.yaml:9]: when the operator is released i think we should update the subscription to use the redhat-operators catalogsource and delete this catalogsource (like we do for ptp and sriov)

Comment (mmirecki) [reply feature-configs/deploy/metallb/operator_catalogsource.yaml:9]: Yes, this is of couse onlyl temporary.
This pointed to the latest master some time ago, I'll update that to latest when merged.

File: cnf-tests/README.md [hunk 5]
```diff
@@ -545,3 +545,4 @@ After running the test suite, all the dangling resources are cleaned up.
 - CLEAN_PERFORMANCE_PROFILE - disable performance profile cleanup for faster tests
 - PERFORMANCE_PROFILE_MANIFEST_OVERRIDE - performance profile manifest override
 - IPERF3_BITRATE_OVERRIDE - set a maximum bitrate for iperf3 to use in ovs_qos tests
+- SKIP_LOCAL_RESOURCES - use default test resource of dependant test suites, using hardcoded defaults instead, needed to successfuly run the metallb e2e tests
```
Comment (SchSeba) [inline cnf-tests/README.md:548]: what will be the difference between true and false here?

does both will work on an openshift cluster?

File: feature-configs/deploy/metallb/operator_catalogsource.yaml (hunk 4 above)
Comment (SchSeba) [reply feature-configs/deploy/metallb/operator_catalogsource.yaml:9]: can't we use the one @sabinaaledort you push to the operator hub? it should be available as part of the 
```
openshift-marketplace   community-operators             Community Operators                     grpc   Red Hat     20d
```
no?

Comment (sabinaaledort) [reply feature-configs/deploy/metallb/operator_catalogsource.yaml:9]: yes, we can definitely use the olm catalogsource.

**Pull request merged**